src/lib/spinel/radio_spinel_impl.hpp

```diff
@@ -1903,20 +1903,21 @@ otError RadioSpinel<InterfaceType, ProcessContextType>::Transmit(otRadioFrame &a
     otPlatRadioTxStarted(mInstance, mTransmitFrame);
 
     error = Request(SPINEL_CMD_PROP_VALUE_SET, SPINEL_PROP_STREAM_RAW,
-                    SPINEL_DATATYPE_DATA_WLEN_S                               // Frame data
-                        SPINEL_DATATYPE_UINT8_S                               // Channel
-                            SPINEL_DATATYPE_UINT8_S                           // MaxCsmaBackoffs
-                                SPINEL_DATATYPE_UINT8_S                       // MaxFrameRetries
-                                    SPINEL_DATATYPE_BOOL_S                    // CsmaCaEnabled
-                                        SPINEL_DATATYPE_BOOL_S                // IsARetx
-                                            SPINEL_DATATYPE_BOOL_S            // SkipAes
-                                                SPINEL_DATATYPE_UINT32_S      // TxDelay
-                                                    SPINEL_DATATYPE_UINT32_S, // TxDelayBaseTime
+                    SPINEL_DATATYPE_DATA_WLEN_S                                   // Frame data
+                        SPINEL_DATATYPE_UINT8_S                                   // Channel
+                            SPINEL_DATATYPE_UINT8_S                               // MaxCsmaBackoffs
+                                SPINEL_DATATYPE_UINT8_S                           // MaxFrameRetries
+                                    SPINEL_DATATYPE_BOOL_S                        // CsmaCaEnabled
+                                        SPINEL_DATATYPE_BOOL_S                    // IsHeaderUpdated
+                                            SPINEL_DATATYPE_BOOL_S                // IsARetx
+                                                SPINEL_DATATYPE_BOOL_S            // SkipAes
+                                                    SPINEL_DATATYPE_UINT32_S      // TxDelay
+                                                        SPINEL_DATATYPE_UINT32_S, // TxDelayBaseTime
                     mTransmitFrame->mPsdu, mTransmitFrame->mLength, mTransmitFrame->mChannel,
                     mTransmitFrame->mInfo.mTxInfo.mMaxCsmaBackoffs, mTransmitFrame->mInfo.mTxInfo.mMaxFrameRetries,
-                    mTransmitFrame->mInfo.mTxInfo.mCsmaCaEnabled, mTransmitFrame->mInfo.mTxInfo.mIsARetx,
-                    mTransmitFrame->mInfo.mTxInfo.mIsSecurityProcessed, mTransmitFrame->mInfo.mTxInfo.mTxDelay,
-                    mTransmitFrame->mInfo.mTxInfo.mTxDelayBaseTime);
+                    mTransmitFrame->mInfo.mTxInfo.mCsmaCaEnabled, mTransmitFrame->mInfo.mTxInfo.mIsHeaderUpdated,
+                    mTransmitFrame->mInfo.mTxInfo.mIsARetx, mTransmitFrame->mInfo.mTxInfo.mIsSecurityProcessed,
+                    mTransmitFrame->mInfo.mTxInfo.mTxDelay, mTransmitFrame->mInfo.mTxInfo.mTxDelayBaseTime);
```

Comment:
This changes the spinel format of `SPINEL_PROP_STREAM_RAW ` (adding an extra boolean in middle of its encoding format) so we need to increment  `SPINEL_RCP_API_VERSION`.

I think this will also cause a backward incompatibility between host and RCP.
- Host that switches to this code won't work anymore with RCP using old format.
- Same situation with RCP's running new code but talking to host that is running older code.

In general, it'd be good to avoid this. But if we need to add backward incompatible change, then  we should ensure to not only incremenent `SPINEL_RCP_API_VERSION` but also `SPINEL_MIN_HOST_SUPPORTED_RCP_API_VERSION` (or need to add code on host to handle spinel frame from older version of RCP).

Reply:
Implemented.